### DISCUSSION TITLE
Delsys sensors names

### DISCUSSION THREAD

**Discussion by M-Hisham** [GiacomoValliPhD/openhdemg#58] [Ideas]
Thank you very much for the great tool
I installed it (after hard time as it is not available on conda yet), and I liked it.

I mainly work on Delsys systems. I wasn't able to open a delsys file using the GUI, as it assumes that the sensor name would be "Galileo sensor", and the force sensor to be 'Trigno Load Cell'

it would be great if both sensors names can be enter using the GUI. so far I changed the package copy on my PC to accommodate my files sensor names, however it would be nice if these information be entered by the user from the GUI

**Comment by GiacomoValliPhD** [comment]
Dear @M-Hisham , thanks for reaching out.

As you might have noticed, the Delsys functionalities have been introduced only a few days ago and have much to improve. We therefore appreciate your feedback. @PaulRitsche is working to make the GUI modular and once it is ready, we will create a window dedicated to tune all the functions parameters. Following your suggestion, we will also allow users to change sensor names in the next releases.

Could you please tell us what sensor you are using? As we expected the current import function to work only with the Galileo sensor.

Stay in touch!

**Reply by M-Hisham** [reply]
Thanks @GiacomoValliPhD for your prompt response. the tool is great and has a lot of potential too.

for Delsys, I am using the Galileo sensor too and the code worked after changing the expected sensor label, however in the Delsys EMG acquisition, we usually name the sensor after the muscle it attached to, or we use any other name to label the sensor.  therefore the sensor_name in the function is more like the sensor name label, but not the actual name of sensor model.

I would be happy to help and test the code too. 
Thanks again

**Reply by GiacomoValliPhD** [reply]
I see, thanks for the explanation! Unfortunately, we cannot know and test all the available recording systems and tools, so we really need your feedback to make them working better!

If you want to perform your analysis using the library instead of the GUI, you should be able to change this parameter in the function's inputs and, currently, calling directly the function instead of relying on the GUI will give you much more flexibility and access to additional features.

If you want to help and test the code, have a look at our contribute section (https://www.giacomovalli.com/openhdemg/contribute/), we are always happy to see people willing to help, especially on the Delsys side, where our experience is limited.

Stay in touch!

**Reply by PaulRitsche** [reply]
@M-Hisham I have integrated the settings in the GUI and you can have a look when pulling the modular_GUI_pr branch. Note, is is still in development and needs to be revised by @GiacomoValliPhD. Otherwise it will be integrated into main in the next weeks. We would greatly appreciate your feedback.

**Comment by PaulRitsche** [comment]
Thanks @M-Hisham  for the feedback. That is indeed a useful feature to add. 
I quickly drafted a new issue for this [Sensor Name DELSYS](https://github.com/GiacomoValliPhD/openhdemg/issues/59), please feel free to comment any relevant information in the thread.
